ompi/mca/coll/base/coll_base_alltoallv.c

```diff
@@ -123,9 +126,11 @@ mca_coll_base_alltoallv_intra_basic_inplace(const void *rbuf, const int *rcounts
  error_hndl:
     /* Free the temporary buffer */
     free (tmp_buffer);
+    if( MPI_SUCCESS != err ) {
+        ompi_coll_base_free_reqs(reqs, 2 );
```

Comment:
@bosilca 
You can come to these instructions if get_reqs() has failed, i.e. reqs=NULL.
So unless you insert this test inside free_reqs(), you should check the pointer here, before calling free_reqs()